docs/x509.rst

```diff
@@ -964,6 +967,15 @@ Exceptions
 
         Returns the OID.
 
+.. class:: UnsupportedGeneralNameType
+
+    This is raised when a certificate contains an unsupported general name
+    type in an extension.
+
+    .. attribute:: type
+
+        :type: :term:`text` or int
```

Comment:
The integer value is standard, yes. It's also explicitly restricted in the RFC to the items we know about in the dict I made. A malicious ASN.1 payload could be constructed that didn't use a valid int though.